dynamic_rest/viewsets.py

```diff
@@ -0,0 +1,31 @@
+from rest_framework import viewsets, response
+
+class DynamicModelViewSet(viewsets.ModelViewSet):
+
+  def _get_request_fields(self):
+    request_fields = self.request.QUERY_PARAMS.getlist('fields[]')
+    field_map = {}
+    for field in request_fields:
+      if field.startswith('-'):
+        field = field[1:]
+        include = False
+      else:
+        include = True
+      field_segments = field.split('.')
+      num_segments = len(field_segments)
+      current_map = field_map
+      for i, segment in enumerate(field_segments):
+        if segment:
```

Comment:
I'm guessing this is so if we get `a..b` we pretend it's `a.b`, but I wonder if it might be better to fail (or ignore the field entirely) since it could be indicative of a bug/issue upstream. Also, weird things might happen if we get `a..` since `a` should be a leaf but won't be treated as such.